ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -353,24 +365,43 @@ writeSnapshot hasFS encLedger encRef ss cs = do
     encode :: ChainSummary l r -> Encoding
     encode = encodeChainSummary encLedger encRef
 
--- | Delete snapshot from disk
-deleteSnapshot :: HasCallStack => HasFS m h -> DiskSnapshot -> m ()
-deleteSnapshot HasFS{..} = removeFile . snapshotToPath
+-- | Delete snapshot from disk, unless it has a suffix
+deleteSnapshot :: (HasCallStack, Monad m) => HasFS m h -> DiskSnapshot -> m ()
+deleteSnapshot HasFS{..} ds@DiskSnapshot {..} =
+    when (isNothing dsSuffix) $ removeFile $ snapshotToPath ds
```

Comment:
We also call it when we fail to use a snapshot at `tryNewestFirst`. Do we still want to delete it in this case case, even though it has a suffix?

Reply:
I see. I don't think we should delete it in that case. I suppose we can do a similar check in `tryNewestFirst`.

It *is* important to do the check in `trimSnapshots`, because I think it can change which snapshots get deleted.

Reply:
> It is important to do the check in trimSnapshots, because I think it can change which snapshots get deleted.

I don't understand this part

Reply:
For example, say you want to keep 2 *non-named* snapshots on disk. You already have two named snapshots that are newer than all non-named snapshot. The LedgerDB just created its third non-named snapshot 140 and now trims it snapshots:
```
1000_shelley
200_favourite_snapshot
140
120
100
```
In `trimSnapshots`, you'll find there are 5 snapshots on disk, now you  drop the first two and try to delete the remaining three: `140`, `120`, and `100`. You end up without non-named snapshots instead of 2.

If you filter first in `trimSnapshots` on "non-named", you'll drop `140` and `120` from the list and delete `100`. Now you end up with 2 non-named snapshots.

Reply:
So we want `onDiskNumSnapshots` to indicate the number of  non-named snapshots? That's not how this pr works